File: retainnonconfigurationinstancemvp/src/main/java/com/example/retainnonconfigurationinstancemvp/retain/DataFragment.java
package com.example.retainnonconfigurationinstancemvp.retain;

import android.app.Fragment;
import android.os.Bundle;

public class DataFragment<Data> extends Fragment {

	private Data data;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);
    }

    public void setData(Data data) {
        this.data = data;
    }

    public Data getData() {
        return data;
    }
}

File: retainFragmentMvp/src/main/java/com/example/antonshevchuk/todolistkotlin/retain/DataFragment.kt
package com.example.antonshevchuk.todolistkotlin.retain

import android.app.Fragment
import android.os.Bundle

class DataFragment<Data> : Fragment() {

	var data: Data? = null

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		retainInstance = true
	}
}